djangosaml2/views.py

```diff
@@ -182,10 +182,10 @@ def login(request,
             logger.warning('IDP %s does not support %s,  trying %s',
                            selected_idp, binding, BINDING_HTTP_POST)
             binding = BINDING_HTTP_POST
-        # if switched binding still not supported, give up
-        if binding not in supported_bindings:
-            raise UnsupportedBinding('IDP %s does not support %s or %s',
-                                     selected_idp, BINDING_HTTP_POST, BINDING_HTTP_REDIRECT)
+    # if switched binding still not supported, give up
+    if binding not in supported_bindings:
+        raise UnsupportedBinding('IDP %s does not support %s or %s',
+                                 selected_idp, BINDING_HTTP_POST, BINDING_HTTP_REDIRECT)
```

Comment:
Might break the code if we hoist the if condition out since it is adding the binding values in the nested if and lastly checking the supported binding value